File: CS/IPAddressEditor/IPAddressHelper.cs
using System;

namespace IPAddressEditor
{
	public class IPAddressHelper
	{
		public static DateTime ToDateTime(IPv4Addr ip)
		{
			byte[] ipParts = ip.ToByteArray();
			string ipStr = "";

			for ( int i = 0; i < ipParts.Length; i++ )
				ipStr += String.Format("{0:d3}", ipParts[i]);

			return new DateTime(Int64.Parse(ipStr));
		}

		public static IPv4Addr ToIPAddress(DateTime dt)
		{
			if ( dt.Ticks == 0 )
				return new IPv4Addr("0.0.0.0");

			string ip = "";

			string strIP = dt.Ticks.ToString();
			while ( strIP.Length < 12 )
				strIP = "0" + strIP;

			while ( strIP != "" )
			{
				ip += Int16.Parse(strIP.Substring(0, 3)).ToString() + ".";
				strIP = strIP.Remove(0, 3);
			}
			ip = ip.Remove(ip.Length - 1);

			return new IPv4Addr(ip);
		}

		public static bool IsConvertible(DateTime dt)
		{
			if ( dt.Ticks > 255255255255 )
				return false;

			return true;
		}
	}
}

File: VB/IPAddressEditor/IPAddressHelper.vb
Imports Microsoft.VisualBasic
Imports System

Namespace IPAddressEditor
	Public Class IPAddressHelper
		Public Shared Function ToDateTime(ByVal ip As IPv4Addr) As DateTime
			Dim ipParts() As Byte = ip.ToByteArray()
			Dim ipStr As String = ""

			For i As Integer = 0 To ipParts.Length - 1
				ipStr &= String.Format("{0:d3}", ipParts(i))
			Next i

			Return New DateTime(Int64.Parse(ipStr))
		End Function

		Public Shared Function ToIPAddress(ByVal dt As DateTime) As IPv4Addr
			If dt.Ticks = 0 Then
				Return New IPv4Addr("0.0.0.0")
			End If

			Dim ip As String = ""

			Dim strIP As String = dt.Ticks.ToString()
			Do While strIP.Length < 12
				strIP = "0" & strIP
			Loop

			Do While strIP <> ""
				ip &= Int16.Parse(strIP.Substring(0, 3)).ToString() & "."
				strIP = strIP.Remove(0, 3)
			Loop
			ip = ip.Remove(ip.Length - 1)

			Return New IPv4Addr(ip)
		End Function

		Public Shared Function IsConvertible(ByVal dt As DateTime) As Boolean
			If dt.Ticks > 255255255255 Then
				Return False
			End If

			Return True
		End Function
	End Class
End Namespace